### DISCUSSION TITLE
Mac, terminal flickers for on going progress

### DISCUSSION THREAD

**Discussion by ankitpatial** [ghostty-org/ghostty#4048] [Q&A] [answered]
Terminal is flickering when there is ongoing task progress bar is updating and terminal itself is not in focus:

https://github.com/user-attachments/assets/52f559f2-14fb-4239-a24a-d31cb49ddcd0

**Comment by mitchellh** [comment, marked as answer]
This is an `ollama` bug, not Ghostty. They need to send [synchronized output](https://gist.github.com/christianparpart/d8a62cc1ab659194337d73e399004036) sequences to prevent tearing.

Ghostty's render loop is going too fast and rendering frames after they erase the line but before they redraw.

**Comment by ankitpatial** [comment]
Thank you for your quick reply and for this wonderful terminal.
I love Ghostty!

**Comment by ttys3** [comment]
come from https://github.com/ghostty-org/ghostty/discussions/5855

it's not `ollama` thing.

all my exists apps, like `top`, `htop` blinks (when run in inactive split)

**Reply by mitchellh** [reply]
Yes we found an alternate reason and I believe it’s fixed on tip. If not, there’s another discussion tracking it now I’ll look for it

**Reply by ttys3** [reply]
> Yes we found an alternate reason and I believe it’s fixed on tip. If not, there’s another discussion tracking it now I’ll look for it

tested latest commit: bd7c5cc95f872d241ddc8aea4c81c540c6d9c19f

the inactive split blink or disappear issue still exists.

**Reply by mitchellh** [reply]
Roger
